drivers/rilmodem/sim.c

```diff
@@ -943,7 +944,8 @@ static void ril_query_pin_retries(struct ofono_sim *sim,
 			g_free(cbd);
 			CALLBACK_WITH_FAILURE(cb, NULL, data);
 		}
-	} else if (sd->vendor == OFONO_RIL_VENDOR_MTK) {
+	} else if (sd->vendor == OFONO_RIL_VENDOR_MTK ||
+			sd->vendor == OFONO_RIL_VENDOR_MTK2) {
```

Comment:
I understand the need for the different quirks, my point was that you're using a "vendor" field for the quirking when the *vendor* hasn't changed.  These are really model ( or family ) specific quirks.  If MTK does yet another re-design, we'd end up with MTK3, which in my mind is less than ideal.  Mind adding a note to the effect that this should be re-factored if/when yet another design lands?